flexget/components/sites/sites/serienjunkies.py

```diff
@@ -27,10 +28,80 @@
 regex_is_subtitle = re.compile(r'Untertitel|Subs?|UT', re.I)
 
 LANGUAGE = ['german', 'foreign', 'subtitle', 'dual']
-HOSTER = ['ul', 'cz', 'so', 'all']
+HOSTER = ['ul', 'so', '1fichier', 'ddl', 'filefactory', 'filer', 'nitroflare', 'oboom', 'rapidgator', 'turbobit',
+          'uploaded', 'share-online', 'all']
 
 DEFAULT_LANGUAGE = 'dual'
-DEFAULT_HOSTER = 'ul'
+DEFAULT_HOSTER = 'uploaded'
+
+HOSTER_MAP = {
+    '1fichier': ['1fichier.com'],
+    'ddl': ['ddl.to'],
+    'filefactory': ['filefactory.com'],
+    'filer': ['filer.net'],
+    'nitroflare': ['nitroflare.com'],
+    'oboom': ['oboom.com'],
+    'rapidgator': ['rapidgator.net'],
+    'turbobit': ['turbobit.net'],
+    'uploaded': ['uploaded.to', 'uploaded.net'],
+    'share-online': ['share-online.to']
+}
+
+
+class SerienjunkiesMatch:
+    def __init__(self):
+        self.hoster = None
+        self.quality = float("inf")
+        self.found = False
+
+    @classmethod
+    def factory(cls, href, next_sibling, hoster_config):
+        instance = cls()
+        if isinstance(hoster_config, str):
+            if hoster_config == 'all' or cls.does_match(href, next_sibling, hoster_config):
+                instance.hoster = hoster_config
+                instance.quality = 0
+                instance.found = True
+            else:
+                instance.quality = float("inf")
+        else:
+            for quality, hoster in enumerate(hoster_config):
+                if cls.does_match(href, next_sibling, hoster):
+                    instance.hoster = hoster
+                    instance.quality = quality
+                    instance.found = True
+                    break
+
+            if not instance.found and 'all' in hoster_config:
+                instance.hoster = 'all'
+                instance.quality = hoster_config.index('all')
+                instance.found = True
+        return instance
+
+    @staticmethod
+    def does_match(href, next_sibling, hoster):
+        match_found = False
+        if hoster == 'ul':
+            hoster = 'uploaded'
+        if hoster == 'so':
+            hoster = 'share-online'
```

Comment:
more pythonic to use dict for this:
```suggestion
		hoster = {'ul': 'uploaded', 'so': 'share-online'}.get(hoster, hoster)
```